File: PhotoBank.ViewModel.Dto/PhotoItemDto.cs
using System;
using System.Collections.Generic;

namespace PhotoBank.ViewModel.Dto
{
    public class PhotoItemDto
    {
        [System.ComponentModel.DataAnnotations.Required]
        public int Id { get; set; }

        [System.ComponentModel.DataAnnotations.Required]
        public required byte[] Thumbnail { get; set; }

        [System.ComponentModel.DataAnnotations.Required]
        public required string Name { get; set; }
        public DateTime? TakenDate { get; set; }
        public bool IsBW { get; set; }
        public bool IsAdultContent { get; set; }
        public double AdultScore { get; set; }
        public bool IsRacyContent { get; set; }
        public double RacyScore { get; set; }
        [System.ComponentModel.DataAnnotations.Required]
        public required string StorageName { get; set; }

        [System.ComponentModel.DataAnnotations.Required]
        public required string RelativePath { get; set; }
        public IEnumerable<TagItemDto>? Tags { get; set; }
        public IEnumerable<PersonItemDto>? Persons { get; set; }
    }
}

File: Photobank.Ts/packages/shared/src/types/dto/PhotoItemDto.ts
import type { TagItemDto } from './TagItemDto';
import type { PersonItemDto } from './PersonItemDto';

export interface PhotoItemDto {
  id: number;
  thumbnail: string;
  name: string;
  takenDate?: string;
  isBW: boolean;
  isAdultContent: boolean;
  adultScore: number;
  isRacyContent: boolean;
  racyScore: number;
  storageName: string;
  relativePath: string;
  tags?: TagItemDto[];
  persons?: PersonItemDto[];
}